packages/ingest/test/test-granule.js

```diff
@@ -339,3 +342,161 @@ test('unversionFilename returns filename without version stamp if present', (t)
 
   t.is(expected, actual);
 });
+
+test('handleDuplicateFile throws DuplicateFile if method is called and duplicateHandling is set to "error"', async (t) => {
```

Comment:
This method wasn't tested at all, so I added tests that cover the logic branches.   Given what that looks like, there's probably room for a refactor to handleDuplicateFile/associated logic, however that's best saved for a PR and *probably* out of scope for this ticket.

Reply:
> ... and ~_probably_~ *_definitely_* out of scope for this ticket.